src/Orleans/Core/GrainAttributes.cs

```diff
@@ -346,18 +347,26 @@ public LogConsistencyProviderAttribute()
                 ProviderName = Runtime.Constants.DEFAULT_LOG_CONSISTENCY_PROVIDER_NAME;
             }
         }
-
     }
 
     [AttributeUsage(AttributeTargets.Class, AllowMultiple = true)]
     public sealed class ImplicitStreamSubscriptionAttribute : Attribute
```

Comment:
Suggest introduction of
abstract class ImplicitStreamSubscriptionBaseAttribute<TPredicate> where TPredicate :  IStreamNamespacePredicate. new()

with ImplicitStreamSubscriptionAttribute : ImplicitStreamSubscriptionBaseAttribute<ExactMatchStreamNamespacePredicate>

So developers can introduce their own attributes with arguments (which simply providing a type does not allow).  For instance
RegExImplicitStreamSubscriptionAttribute : ImplicitStreamSubscriptionBaseAttribute<RegExMatchStreamNamespacePredicate> with constructors that take a RegEx expression.